session/pingpong/consumer_balance_tracker.go

```diff
@@ -144,7 +147,32 @@ func (cbt *ConsumerBalanceTracker) handleTopUpEvent(id string) {
 func (cbt *ConsumerBalanceTracker) handleRegistrationEvent(event registry.RegistrationEventPayload) {
 	switch event.Status {
 	case registry.RegisteredConsumer, registry.RegisteredProvider:
-		cbt.updateBalanceFromAccountant(event.ID)
+		var boff backoff.BackOff
+		eback := backoff.NewExponentialBackOff()
+		eback.MaxElapsedTime = time.Minute
+		eback.InitialInterval = time.Second * 2
+
+		boff = backoff.WithMaxRetries(eback, 10)
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		go func() {
+			defer cancel()
+			select {
+			case <-cbt.stop:
+			case <-ctx.Done():
+			}
+		}()
+
+		boff = backoff.WithContext(boff, ctx)
+		toRetry := func() error {
+			return cbt.updateBalanceFromAccountant(event.ID)
```

Comment:
What if registration success event is send to node only after balance is ready?

Reply:
In that case, we'd not need the retry here.